hibernate-core/src/main/java/org/hibernate/internal/CacheImpl.java

```diff
@@ -402,17 +402,18 @@ public Region getNaturalIdCacheRegion(String regionName) {
 		// which Map contents are added to getAllSecondLevelCacheRegions.
 		// In addition, if there is a CollectionRegion and an EntityRegion with the same name, then we
 		// want the EntityRegion to be in the Map that gets returned.
-		final Map<String, Region> allCacheRegions = new HashMap<String, Region>(
-				2 + queryCaches.size() + otherRegionMap.size() + naturalIdRegionMap.size()
-						+ collectionRegionMap.size() + entityRegionMap.size()
-		);
-		allCacheRegions.put( updateTimestampsCache.getRegion().getName(), updateTimestampsCache.getRegion() );
-		allCacheRegions.put( queryCache.getRegion().getName(), queryCache.getRegion() );
+		final Map<String, Region> allCacheRegions = new HashMap<String, Region>();
```

Comment:
@rvansa, is it worthwhile to instantiate the ```HashMap``` to it's ultimate size, since it can be computed beforehand?

AFAIK, Hibernate only uses ```getNaturalIdCacheRegion``` when generating statistics. An application could use this method as well.

Reply:
@gbadner I am not a profiler :) IMO this is not a perf-critical method, I'd personally rather keep it simple.

Reply:
@rvansa, thanks.